src/Orchard.Cms.Web/Modules/Orchard.OpenId/Views/Admin/Edit.cshtml

```diff
@@ -132,145 +118,147 @@
+            else {
+                clientSecretHints.each(function () { this.innerText = this.innerText.replace("client_id, client_secret, ", "client_id, "); });
+            }
         }
-    }
 
 
-    $("#AllowRefreshTokenFlow").change(function () {
-        refreshOfflineAccessTip();
-    });
+        $("#AllowRefreshTokenFlow").change(function () {
+            refreshOfflineAccessTip();
+        });
 
-    function refreshOfflineAccessTip(defaultValue) {
-        var offlineAccessHints = $("#AllowPasswordFlowRecommendedHint, #AllowAuthorizationCodeFlowRecommendedHint, #AllowHybridFlowRecommendedHint");
-        var allowRefreshTokenFlow = $("#AllowRefreshTokenFlow");
-        if (defaultValue === allowRefreshTokenFlow.prop('checked'))
-            return;
+        function refreshOfflineAccessTip(defaultValue) {
+            var offlineAccessHints = $("#AllowPasswordFlowRecommendedHint, #AllowAuthorizationCodeFlowRecommendedHint, #AllowHybridFlowRecommendedHint");
+            var allowRefreshTokenFlow = $("#AllowRefreshTokenFlow");
+            if (defaultValue === allowRefreshTokenFlow.prop('checked'))
+                return;
 
-        if (allowRefreshTokenFlow.prop('checked')) {
-            offlineAccessHints.each(function () { this.innerText = this.innerText.replace("roles", "roles, offline_access"); });
-        }
-        else {
-            offlineAccessHints.each(function () { this.innerText = this.innerText.replace(", offline_access", ""); });
+            if (allowRefreshTokenFlow.prop('checked')) {
+                offlineAccessHints.each(function () { this.innerText = this.innerText.replace("roles", "roles, offline_access"); });
+            }
+            else {
+                offlineAccessHints.each(function () { this.innerText = this.innerText.replace(", offline_access", ""); });
+            }
         }
-    }
-
-    $("#AllowClientCredentialsFlow, #AllowPasswordFlow, #AllowAuthorizationCodeFlow, #AllowHybridFlow, #AllowImplicitFlow, #AllowRefreshTokenFlow").change(function () {
-        refreshFlows();
-    });
 
-    function refreshFlows() {
-        refreshRoleGroup();
-        refreshAllowRefreshTokenFlowVisibility();
-        refreshRedirectSettings();
-    }
-    function refreshRoleGroup() {
-        $("#RoleGroup").collapse($("#AllowClientCredentialsFlow").prop('checked') ? "show" : "hide");
-    }
-    function refreshAllowRefreshTokenFlowVisibility() {
-        var allowRefreshTokenFlow = $("#AllowRefreshTokenFlow");
+        $("#AllowClientCredentialsFlow, #AllowPasswordFlow, #AllowAuthorizationCodeFlow, #AllowHybridFlow, #AllowImplicitFlow, #AllowRefreshTokenFlow").change(function () {
+            refreshFlows();
+        });
 
-        if (($("#AllowPasswordFlow").prop('checked') || $("#AllowAuthorizationCodeFlow").prop('checked') || $("#AllowHybridFlow").prop('checked'))) {
-            allowRefreshTokenFlow.removeAttr("disabled");
+        function refreshFlows() {
+            refreshRoleGroup();
```

Comment:
Roles is a legacy concept for authorization and it shouldn´t be used for authorize but it doesn't means it is used for grouping claims. As actually Orchard uses it. I don't think we need to inform of roles in access token, what we need is to provide claims related to roles assigned to an app. Like we did in oauth branch here:
https://github.com/OrchardCMS/Orchard2/blob/oauth2/src/Orchard.Cms.Web/Modules/Orchard.OpenId/Services/OpenIdUserManager.cs#L37
and here: https://github.com/OrchardCMS/Orchard2/blob/oauth2/src/Orchard.Cms.Web/Modules/Orchard.OpenId/Services/OpenIdApplicationManager.cs#L78
So, if do you agree I will undo your changes on UI for setting roles for client credentials flows and I will add here https://github.com/OrchardCMS/Orchard2/blob/openidUpgrade/src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs#L334 code for adding claims related with roles assigned to an app.